mkinitcpio

```diff
@@ -211,25 +211,33 @@ build_image() {
             ;;
     esac
 
-    cpio_opts=('-0' '-o' '-H' 'newc')
-    (( _optquiet )) && cpio_opts+=('--quiet')
+    cpio_opts=('--null' '-cf-' '--format=newc')
+    (( _optquiet )) || cpio_opts+=('-v')
     if (( EUID != 0 )); then
         warning 'Not building as root, ownership cannot be preserved'
-        cpio_opts+=('-R' '0:0')
+        cpio_opts+=('--uid' '0' '--gid' '0')
```

Comment:
We should probably unconditionally do this if we're committing to reproducible builds. There's little value in permissions in early userspace. There might not even be an /etc/passwd or /etc/group to map numeric IDs back to names.

Reply:
Agreed.